spec.html

```diff
@@ -124,7 +140,8 @@ <h1>get ResizableArrayBuffer.prototype.byteLength</h1>
         1. Let _O_ be the *this* value.
         1. Perform ? RequireInternalSlot(_O_, [[ArrayBufferMaxByteLength]]).
         1. If IsDetachedBuffer(_O_) is *true*, return *+0*<sub>𝔽</sub>.
-        1. Let _length_ be ArrayBufferByteLength(_O_).
+        1. NOTE: The ~SeqCst~ argument below is ignored for ResizableArrayBuffer.
+        1. Let _length_ be ArrayBufferByteLength(_O_, ~SeqCst~).
```

Comment:
```suggestion
        1. Let _length_ be ! ArrayBufferByteLength(_O_, ~SeqCst~).
```
(and throughout)

Reply:
I prefer no `!` here. This is not a user code-visible AO and I don't think it's confusingly named in the same that `MakeIdempotentArrayBufferByteLengthGetter` is.

Reply:
It’s certainly an editorial call; but the point of the notation is that *everything* (modulo predicates) has a ? or a ! precisely so i don’t have to think about whether it is a “user code-visible AO” - how else can i know that without looking it up?

Reply:
Yeah, it's certainly been a longstanding point of disagreement between us. My mental model is:

- If calling an AO() doesn't have ? or !, then it either doesn't return Completion Records, or it does and the completion is manually handled. Which one it does should be easy to see by local inspection of how the return value is used. If it doesn't return Completion Records, obviously it can't throw.
- If it has a ?, then the AO returns Completion Records and can throw.
- If it has a !, then the AO returns Completion Records and cannot throw.

Reply:
At the moment, every AO returns a completion record.

Reply:
I know, I know, and I hate it and it's kind of nonsensical.